src/LibraryManager/Providers/Shared/ProvidersCommon.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.Web.LibraryManager.Contracts;
+
+namespace Microsoft.Web.LibraryManager.Providers.Shared
+{
+    internal static class ProvidersCommon
+    {
+        private const char _idPartsSeparator = '@';
+
+        public static LibraryIdentifier GetLibraryIdentifier(IProvider provider, string libraryId)
+        {
+            // A valid libraryId:
+            // - can not be null or empty string
+            // - has at least one _idPartsSeparator
+            // - can not end with a _idPartsSeparator
+            // - can start with a _idPartsSeparator
+            // - can not start or end with space
+            // - each part (Name, Version) can not start or end with space 
+
+            if (string.IsNullOrEmpty(libraryId) ||
+                !libraryId.Contains(_idPartsSeparator.ToString()) ||
```

Comment:
>.Contains(_idPartsSeparator.ToString() [](start = 26, length = 38)

Why not use the char versions of these calls? IE,

...
(libraryId.IndexOf(_idPartsSeparator) < 0) ||
(libraryId[libraryId.Length - 1] == idPartsSeparator) ||
...